uamqp/client.py

```diff
@@ -946,8 +949,13 @@ def _client_run(self):
             if self._timeout > 0:
                 timespan = now - self._last_activity_timestamp
                 if timespan >= self._timeout:
-                    _logger.info("Timeout reached, closing receiver.")
-                    self._shutdown = True
+                    self._timeout_reached = True
+                    if self._shutdown_after_timeout:
+                        _logger.info("Timeout reached, closing receiver.")
+                        self._shutdown = True
+                    else:
+                        self._last_activity_timestamp = None # To reuse the receiver, reset the timestamp
```

Comment:
This comment is excellent (in terms of value added) but begs the question: How would one be expected to do this in normal operation?

(My understanding, to doubly use this to ensure I'm on the right page: ".  This can be achieved by performing a do_work loop until client_ready is true."  although feel free to be more terse if that can be summarized better.)

Reply:
I'm assuming "do this" in your question refers to `resetting self._last_activity_timestamp to None` and "normal operation" refers to all the receiving mechanisms Anna mentioned in the above comment.

so the answer to your question is that users doesn't have control over the resetting in current implementation, Timeout reaches in `do_work` is the only trigger to reset it.

But your understanding of the question makes me wonder if the "reset" logic should be more explicit -- just as how you understand, fetching a new receive_messages_iter would do the reset.

Another option would be to reset `self._last_activity_timestamp` proactively every time we get a new iterator, no longer triggered by timeout, and we need to this for all of our receiving mechanisms as mentioned by Anna.

Reply:
I need to write some code to feel which option is better and taking all of our mechanisms into consideration.
feel free to leave your thoughts.

Reply:
@KieranBrantnerMagee , I think you're right, I should be more explicit on this.
I've added "reset self._last_activity_timestamp" logic in each receive mechanism, matching your understanding now.

Reply:
I should have responded back sooner to provide clarity, but I think this actually ended up for the best, because my head wasn't entirely fully "together' on what the correct comment should say, and your takeaway sounds far more well-thought-out than what I had in mind.  Will read the implementation with this context.